sdk/src/main/java/com/hedera/hashgraph/sdk/Query.java

```diff
@@ -138,6 +143,54 @@ boolean isPaymentRequired() {
 
     abstract void validateChecksums(Client client) throws BadEntityIdException;
 
+    Client.Operator getOperatorFromClient(Client client) {
+        var operator = client.getOperator();
+
+        if (operator == null) {
+            throw new IllegalStateException(
+                "`client` must have an `operator` or an explicit payment transaction must be provided");
+        }
+
+        return operator;
+    }
+
+    @Override
+    void onExecute(Client client) throws Exception {
+        initWithNodeIds(client);
```

Comment:
Same comment about duplication. Should do something similar to how you avoided the duplication in `Transaction.onExecute()`.